lib/EspWire/src/esp_twi.cpp

```diff
@@ -160,6 +168,8 @@ static inline IRAM_ATTR unsigned int _getCycleCount()
     unsigned int ccount = 0;
     __asm__ __volatile__("esync; rsr %0,ccount":"=a" (ccount));
     return ccount;
+#elif defined(ESP32_C3)
+  //what is this??
```

Comment:
according to the [comparison sheet](https://docs.espressif.com/projects/esp-idf/en/v4.3.1/esp32/hw-reference/chip-series-comparison.html) C3 does not have a Pulse counter

Reply:
we can use `ESP.getCycleCount()` or `cpu_hal_get_cycle_count()`
https://github.com/espressif/arduino-esp32/blob/e0657db0f0d4cfe9320744faf2507123790cf56c/cores/esp32/Esp.h#L114 without this I2C timinigs will be wrong. It might explain baro detecting issues, because without it bus works with highest possible speed. This has to be verified and tuned with oscilloscope or logic analyzer later to ensure that I2C bus works correctly.

I suppose the `ESP32_C3` define comes from Arduino. I would expect it to be rather `ESP32C3`, but I haven't verified it yet.

Reply:
and actually `ccount` it is not `pulse counter`, it is an internal cycle counter that is incremented on every instruction. Did you get a compiler error here?

Reply:
yes, asm code trows error here

Reply:
fixed from https://github.com/junhuanchen/esp-idf-software-serial/commit/1e5b491abe166db66fd12422e39a838b0727d35b